tree/tree/src/TBranch.cxx

```diff
@@ -1878,6 +1878,22 @@ TBasket *TBranch::GetFreshCluster()
    return basket;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Return the 'full' name of the branch.  In particular prefix  the mother's name
+/// when it does not end in a trailing dot and thus is not part of the branch name
+TString TBranch::GetFullName() const
+{
+   TBranch* mother = GetMother();
+   if (!mother || mother==this) {
+      return fName;
+   }
+   TString motherName(mother->GetName());
```

Comment:
Shouldn't this be `mother->GetFullName()`?

Reply:
It should not matter.  Since the mother is the top level node (i.e. the ultimate parent) the full name and the name should be identical.